### DISCUSSION TITLE
LSP functionality for creating the configuration file

### DISCUSSION THREAD

**Discussion by rchl** [biomejs/biome#1007] [Ideas]
The `biome init` command creates a `biome.json` configuration file. My suggestion is to provide the equivalent functionality through LSP.

Why? Because running `biome init` in context of LSP, when running in an editor, is not straightforward. There is no guarantee that the project is running a local `biome` binary (the editor might be running bundled `biome` binary) so the user might not have easy access to it. And even when having `biome` installed locally in the project, the command for invoking `biome init` can differ depending on the package manager used so it's not easy for the editor integration to provide convenient command for triggering `init`.

Thus I suggest for biome LSP layer to provide a corresponding functionality so that it's possible to create the configuration file that way.

I can think of 3 ways of how this could be done (spoiler alert - the third one is probably the most sensible):

1. Expose a custom [LSP command](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_executeCommand) that the client could invoke. The server would then provide a [workspace edit with the CreateFile operation](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#resourceChanges) that the editor would handle.
2. Have the biome LSP server create the file itself in the root of the workspace. Though that would IMO not be LSP-like since the LSP might in theory not have access to the underlying file system.
3. Yet another alternative would be for the biome LSP to provide a "source" [code action](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_codeAction). Those code actions only show up in the editor when explicitly invoking "source actions" command. The advantage of that over a custom command is that there is no extra logic needed for the client to implement. In fact I think that's the best option.

(Same suggestion can apply to the `migrate` command)

**Comment by ematipico** [comment]
That sounds like a good suggestion!
